### ISSUE TITLE
Token Problem

### ISSUE THREAD

**Issue by venisassist** [stackblitz/bolt.new#1147]
### Describe the bug

First of all, you have done a great job, congratulations. However, while working on the VenisAI application, we encountered an error and we could not fix this error, and my 3.5 million tokens were used up for nothing. So, I used 3.5 million tokens to fix the error, but the error was not fixed. I ask for your support in this matter. I also have a suggestion for this beautiful project of yours. It would be better if the token is not lost in case of errors encountered during operation.

### Link to the Bolt URL that caused the error

https://bolt.new/~/sb1-h3e2le

### Steps to reproduce

I ask for your support in this matter. I also have a suggestion for this beautiful project of yours. It would be better if the token is not lost in case of errors encountered during operation.

### Expected behavior

I ask for your support in this matter. I also have a suggestion for this beautiful project of yours. It would be better if the token is not lost in case of errors encountered during operation.

### Platform

- OS: [e.g. macOS, Windows, Linux]
- Browser: [e.g. Chrome, Safari, Firefox]
- Version: [e.g. 91.1]

**Comment by venisassist** [comment]
First of all, you have done a great job, congratulations. However, while working on the VenisAI application, we encountered an error and we could not fix this error, and my 3.5 million tokens were used up for nothing. So, I used 3.5 million tokens to fix the error, but the error was not fixed. I ask for your support in this matter. I also have a suggestion for this beautiful project of yours. It would be better if the token is not lost in case of errors encountered during operation.
